gensim/models/atmodel.py

```diff
@@ -767,9 +767,9 @@ def update(self, corpus=None, author2doc=None, doc2author=None, chunksize=None,
 
             # Train on all documents of authors in input_corpus.
             train_corpus_idx = []
-            for _ in author2doc.keys():  # For all authors in input corpus.
-                for doc_ids in self.author2doc.values():  # For all documents in total corpus.
-                    train_corpus_idx.extend(doc_ids)
+            # Collect all documents of authors.
+            for doc_ids in self.author2doc.values():
+                train_corpus_idx.extend(doc_ids)
```

Comment:
Why keep `train_corpus_idx` as `list` in the first place? It looks like it's converted to a set right below, so maybe better to make it a `set` from the start?

Reply:
That is true, indeed. I would propose something like this:
```
# Train on all documents of authors in input_corpus.
train_corpus_idx = set()
# Collect all documents of authors.
for doc_ids in self.author2doc.values():
    train_corpus_idx.update(doc_ids)

train_corpus_idx = list(train_corpus_idx)
```

Reply:
I'm not familiar with the algo, but for the sake of reproducibility, I'd replace `list(train_corpus_idx)` with `sorted(train_corpus_idx)`. That will remove randomness from the ordering of values, while still producing a list.